### PR TITLE
`General`: Add course title to enrollment confirmation

### PR DESCRIPTION
**Pull request by milljoniaer**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client

- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
This PR adds the course Title to the modal title of the Course Enrollment confirmation popup as requested by Stephan.

### Steps for Testing

Prerequisites:
- 1 Student 
- 1 Course with active Course Enrollment and Confirmation Message 

1. Log in to Artemis
2. Navigate to Course Enrollment
3. Enroll for this course
4. Verify that there is the course title within the Modal title

### Screenshots
image

### CODE REVIEW

File: src/main/webapp/app/overview/course-registration/course-registration-button/course-registration-button.component.ts [hunk 1]
```diff
@@ -17,11 +18,14 @@ export class CourseRegistrationButtonComponent implements OnInit {
     userIsAllowedToRegister = false;
     loading = false;
 
+    confirmationTitle: string;
```
Comment (JohannesStoehr) [inline src/main/webapp/app/overview/course-registration/course-registration-button/course-registration-button.component.ts:21]: Am I missing something? You only add this variable and initialize it, but I cannot see where you use its value

**Comment by egekurt123** [comment]
Screenshot 2023-10-22 194508
Currently, the pop-up does not showcase the course title

**Comment by milljoniaer** [comment]
Hi, thank you for your reviews, I recognized that there is a better way to do this. 
So I changed the strategy, please have another look.

I also added a screenshot, who the modal should look like now.

**Review by jakubriegel — APPROVED** [review]
Code ok. Change works on TS4

**Review by sarpsahinalp — COMMENTED** [review]
I wasn't able to see this feature on ts2 for the enrollment test course.
image

On ts3 it seems to be working.
image

**Review by JanaNF — APPROVED** [review]
Manual tested on legacy ts2. Worked as expected.

**Review by LenaKahle — APPROVED** [review]
Tested on ts4, works as intended. 
Initially I thought the green success message ("Enrollment successful") should contain the module name as well, but I guess it's intended only for the confirmation popup.

**Pull request merged**